examples/mp3.go

```diff
@@ -0,0 +1,64 @@
+package main
+
+import (
+	"bytes"
+	"encoding/binary"
+	"fmt"
+	"os"
+	"os/signal"
+
+	"github.com/bobertlo/go-mpg123/mpg123"
+	"github.com/gordonklaus/portaudio"
+)
+
+func main() {
+	if len(os.Args) < 2 {
+		fmt.Println("missing required argument:  input file name")
+		return
+	}
+	fmt.Println("Playing.  Press Ctrl-C to stop.")
+
+	sig := make(chan os.Signal, 1)
+	signal.Notify(sig, os.Interrupt, os.Kill)
+
+	// create mpg123 decoder instance
+	decoder, err := mpg123.NewDecoder("")
+	chk(err)
+
+	fileName := os.Args[1]
+	chk(decoder.Open(fileName))
+	defer decoder.Close()
+
+	// get audio format information
+	rate, channels, encoding := decoder.GetFormat()
+
+	// make sure output format does not change
+	decoder.FormatNone()
+	decoder.Format(rate, channels, encoding)
+
+	portaudio.Initialize()
+	defer portaudio.Terminate()
+	out := make([]int32, 8192)
+	stream, err := portaudio.OpenDefaultStream(0, channels, float64(rate), len(out), &out)
+	chk(err)
+	defer stream.Close()
+
+	chk(stream.Start())
+	defer stream.Stop()
+	for {
+		audio := make([]byte, 2*len(out))
```

Comment:
`2*len(out)` is half as many bytes as are necessary to fill `out`.  I'd expect you to hear some pretty bad audio artifacts as a result of the the latter half of `out` containing silence.